### PR TITLE
🐛 Use rpc proxy in frontend

### PR DESCRIPTION
**Pull request by williamchong**

### CODE REVIEW

File: server/api/index.js [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
-    const parts = req.url.split('?');
-    const queryString = parts[1];
-    const updatedPath = `/txs${parts[0]}`;
-    return updatedPath + (queryString ? `?${queryString}` : '');
-  },
+let proxyPath = '';
+try {
+  const urlObj = new URL(COSMOS_RPC_ENDPOINT);
+  proxyPath = urlObj.pathname;
+  proxyPath = proxyPath.slice(0, proxyPath.length - 1);
```
Comment (WeiJun0827) [inline server/api/index.js:13]:
```suggestion
  const { pathname } = urlObj;
  proxyPath = pathname.slice(0, pathname.length - 1);
```
Would this be more readable?

File: server/util/cosmos.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ import { PUBSUB_TOPIC_MISC } from '../constant';
 import publisher from './gcloudPub';
 
 import {
-  COSMOS_ENDPOINT as cosmosRPCEndpoint,
+  COSMOS_RPC_ENDPOINT as cosmosRPCEndpoint,
```
Comment (WeiJun0827) [inline server/util/cosmos.js:23]:
```suggestion
  COSMOS_RPC_ENDPOINT,
```
Maybe could skip the rename step and directly use `COSMOS_RPC_ENDPOINT`  just like `COSMOS_GAS_PRICE` and other config parameters?

File: server/api/index.js (hunk 1 above)
Comment (WeiJun0827) [inline server/api/index.js:13]:
```suggestion
  const { pathname } = urlObj;
  proxyPath = pathname.slice(0, pathname.length - 1);
```
Would this be more readable?

File: server/util/cosmos.js (hunk 2 above)
Comment (WeiJun0827) [inline server/util/cosmos.js:23]:
```suggestion
  COSMOS_RPC_ENDPOINT,
```
Prefer to skip the rename step and just use `COSMOS_RPC_ENDPOINT` as the parameter like the rest of the config parameters.

File: server/config/config.js [hunk 3]
```diff
@@ -3,7 +3,8 @@ export const FIRESTORE_CONFIG_ROOT = '';
 
 export const ETH_CONTRACT_ADDRESS = '0x1111111111111111111111111111111111111111';
 
-export const COSMOS_ENDPOINT = 'http://LIGHT_CLIENT_ENDPOINT/';
+export const COSMOS_LCD_ENDPOINT = 'http://LIGHT_CLIENT_ENDPOINT/';
```
Comment (WeiJun0827) [inline server/config/config.js:6]: Do we still need to keep the `COSMOS_LCD_ENDPOINT`? Seems no files are going to use it?

**Comment by williamchong** [comment]
rpc connectivity issue in frontend, close for now

**Pull request closed without merging**